src/miner.cpp

```diff
@@ -318,6 +330,16 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
             coinbaseTx.vout[0].nValue = CalculateCoinbaseReward(blockReward, consensus.dist.masternode);
         }
 
+        if (nHeight >= consensus.NextNetworkUpgradeHeight && !evmHeader.empty()) {
```

Comment:
Let's use `IsEvmEnabled` and move the fork check into `IsEVMEnabled` for all uses